lib/imagery/sig.c

```diff
@@ -204,6 +211,12 @@ int I_read_signatures(FILE * fd, struct Signature *S)
         return -1;
     }
 
+    /* Read marker of original class value presence */
+    if (ver >= 2 && fscanf(fd, "%d", &S->have_oclass) != 1) {
+        G_warning(_("Invalid signature file"));
```

Comment:
The message is too generic while there is context available. Please add info on what went wrong, e.g., `Invalid signature file: Original class value presence not readable`